ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -9384,6 +9384,12 @@ IHqlExpression * HqlCppTranslator::getResourcedGraph(IHqlExpression * expr, IHql
     if (true)
         resourced.setown(optimizeCompoundSource(resourced, CSFpreload|csfFlags));
 
+    //Check to see if fields can be removed - this helps LOOP bodies, but also seems to help situations where hoisting
+    //expressions prevents child expressions from preventing fields from being removed.
+    //Perform before the optimizeHqlExpression() so decisions about reducing row sizes are accurate
+    traceExpression("BeforeImplicitProjectGraph", resourced);
+    resourced.setown(insertImplicitProjects(*this, resourced, false));
```

Comment:
@shamser is this a comment on the pull request, or a general comment?  Describing how insertImplictProjects works is outside the scope of this change - you could open a jira.

If you're happy with the PR you should reassign to richard.